ol-keycloak/ol-spi/src/main/java/edu/mit/keycloak/admin/ui/OLSettingsAdminUiTab.java

```diff
@@ -0,0 +1,82 @@
+package edu.mit.keycloak.admin.ui;
+
+import org.keycloak.Config;
+import org.keycloak.component.ComponentModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.KeycloakSessionFactory;
+import org.keycloak.models.RealmModel;
+import org.keycloak.provider.ProviderConfigProperty;
+import org.keycloak.provider.ProviderConfigurationBuilder;
+import org.keycloak.services.ui.extend.UiTabProvider;
+import org.keycloak.services.ui.extend.UiTabProviderFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import edu.mit.keycloak.util.OLAttributeKeys;
+/**
+ * Admin UI Tab for providing additional realm-level settings for MIT Open.
+ *
+ * These are things that don't fit well under different scopes like clients.
+ *
+ */
+public class OLSettingsAdminUiTab implements UiTabProvider, UiTabProviderFactory<ComponentModel> {
+    public static String ID = "Open Learning";
+
+    @Override
+    public String getId() {
+        return ID;
+    }
+
+    @Override
+    public String getHelpText() {
+        return "Configuration specific to Open Learning";
+    }
+
+    @Override
+    public void init(Config.Scope config) {
+    }
+
+    @Override
+    public void postInit(KeycloakSessionFactory factory) {
+    }
+
+    @Override
+    public void close() {
```

Comment:
Pressing cancel on this page while the Canoncial Home URL field is empty results in an error.

Reply:
I haven't been able to figure out what's causing this to break, I also wasn't able to figure out where the code for the button is, but it's likely in the admin theme (react based). I'd say this has been enough of a time sync when in deployed environments ol-infrastructure is going to set it and the UI itself won't be used.